jenkins/xml/node-test-commit-custom-suites-freestyle.xml

```diff
@@ -0,0 +1,208 @@
+    <extensions>
+      <hudson.plugins.git.extensions.impl.CleanCheckout/>
+      <hudson.plugins.git.extensions.impl.CloneOption>
+        <shallow>false</shallow>
+        <noTags>true</noTags>
+        <reference>/data/iojs/git/io.js.reference</reference>
+        <timeout>20</timeout>
+        <depth>0</depth>
+        <honorRefspec>false</honorRefspec>
+      </hudson.plugins.git.extensions.impl.CloneOption>
+      <org.nodejs.jenkinsplugins.nodeversion.NodeVersionExtension plugin="node-version-jenkins-plugin@1.0-SNAPSHOT">
+        <exportNodejsVersion>true</exportNodejsVersion>
+      </org.nodejs.jenkinsplugins.nodeversion.NodeVersionExtension>
+    </extensions>
+  </scm>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers/>
+  <concurrentBuild>true</concurrentBuild>
+  <customWorkspace>node-test-commit-custom-suites</customWorkspace>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>curl https://raw.githubusercontent.com/nodejs/build/master/jenkins/scripts/node-test-commit-pre.sh | bash -ex -s before</command>
+    </hudson.tasks.Shell>
+    <hudson.tasks.Shell>
+      <command>#/bin/bash
+
+echo $NODE_NAME
+
+rm -rf build
+git clone https://github.com/nodejs/build.git
+. ./build/jenkins/scripts/select-compiler.sh
+
+SKIP_MESSAGE=&quot;ok&quot;
+REVELANT_VERSION=1
+if [ &quot;$NODEJS_MAJOR_VERSION&quot; -lt &quot;6&quot; ]; then
+  REVELANT_VERSION=
+  # 390 is only supported on v6 and later for now
+  SKIP_MESSAGE=&quot;ok # skipped s390 not supported for versions less than 6, skipping&quot;
+fi
+
+RELEVANT_SUITE=1
+if [ &quot;$CI_JS_SUITES&quot; = &quot;v8-updates&quot; ] &amp;&amp; [ &quot;$NODEJS_MAJOR_VERSION&quot; -lt &quot;11&quot; ]; then
+  RELEVANT_SUITE=
+fi
+
+if [ &quot;$TEST_ARGS&quot; = &quot;--worker&quot; ] &amp;&amp; [ &quot;$NODEJS_MAJOR_VERSION&quot; -lt &quot;10&quot; ]; then
+  RELEVANT_SUITE=
+fi
+
+JOBS=$(getconf _NPROCESSORS_ONLN)
+
+if [ $REVELANT_VERSION ] &amp;&amp; [ $RELEVANT_SUITE ]; then
+  NODE_TEST_DIR=${HOME}/node-tmp
+  NODE_COMMON_PORT=15000
+  PYTHON=python
+  FLAKY_TESTS=$FLAKY_TESTS_MODE
+  make build-ci -j $JOBS
+  python tools/test.py -j $JOBS -p tap --logfile test.tap --mode=release --flaky-tests=dontcare $TEST_ARGS $CI_JS_SUITES
+else
+  # fake out so we don&apos;t get failures
+  mkdir out
+  echo &quot;1..1&quot; &gt;test.tap
+  echo &quot;ok 1 # skip $SKIP_MESSAGE&quot; &gt;&gt; test.tap
+  echo &quot;  ---&quot; &gt;&gt; test.tap
+  echo &quot;  duration_ms: 4.925&quot; &gt;&gt; test.tap
+  echo &quot;  ...&quot; &gt;&gt; test.tap
+fi</command>
+    </hudson.tasks.Shell>
+    <hudson.tasks.Shell>
+      <command>curl https://raw.githubusercontent.com/nodejs/build/master/jenkins/scripts/node-test-commit-diagnostics.sh | bash -ex -s after</command>
+    </hudson.tasks.Shell>
+    <hudson.tasks.Shell>
```

Comment:
I guess it's okay here, but isn't this usually part of the "post build actions" section in the UI (just for consistency)?

Reply:
I figured recently that it wasn't optimal to put it as a post-build action. I want to start moving them all to a last build stage.
But maybe it should be before the diags-after...
Also it could wait for a future iteration.